src/query_backend.h

```diff
@@ -1058,6 +1075,14 @@ class Sha1 {
     }
   }
 
+  inline void Update(const std::map<std::string, std::map<std::string, double>>& x) {
```

Comment:
This function would be improved by a docstring explaining what it's doing (especially since it has such a vague name.)

Reply:
Nevermind, I see now that this is the pattern. 

However, notably, your function is more complex than the others. I wonder if perhaps a little documentation regarding why this update function works the way it does ... would be helpful.

Reply:
To be honest, I'm also not sure what it does but it is done for each typesystem and so i followed the pattern @katyhuff

Reply:
The function that I added is copied from the type system `mapstringmapstringint` in line 1053 onwards

Reply:
those Update method don't have a docstring... there is a small one line 715...